packages/repository/src/entity/deepReadonly.test.ts

```diff
@@ -78,4 +78,16 @@ describe("Deep Readonly", () => {
       readonlyObj.bar.foo = "baz"
     }).toThrow()
   })
+
+  it("Given an object with function, When deepReadonly is called, Then return the same object with function", () => {
+    const obj = {
+      func: () => "baz",
+    }
+
+    const readonlyObj = deepReadonly(obj)
+
+    expect(readonlyObj).toBe(obj)
+    expect(readonlyObj.func).toBe(obj.func)
+    expect((readonlyObj.func as () => string)()).toBe("baz")
```

Comment:
Te rzutowanie pokazuje raczej błąd w typie, który trzeba załatać - funkcja w obiekcie powinna być dalej normalnie wywoływana.  Musi być omijana przez Readonly.

Reply:
ok, zobacz teraz.

Reply:
Ok, funkcjonalnie już ok 👍  
Ale można popracować nad typem `DeepReadonly` ;) Przyjrzyj mu się - jak przeczytasz linijka po linijce na głos jakbyś tłumaczył do kaczuszki - to myśle że znajdziesz już miejsce w którym wykluczamy pewne elementy, których również nie wrapujemy w <DeepReadonly> ;)

Reply:
Można też zrobić dla czytelności np  ale to raczej zbędne
<img width="637" alt="Zrzut ekranu 2024-02-9 o 19 51 27" src="https://github.com/archisquad/repository/assets/79842760/4f451065-9316-4ed5-b4b7-db975a65ef58">